runtime/runtime.h

```diff
@@ -20,29 +32,17 @@ typedef enum {
 } AccessFlags;
 
 Context createDefaultContext();
-Context createContext(const char* platform_subname, const char* device_type_str);
 void destroyContext(Context ctx);
 void waitFinished(Context ctx);
 
 Buffer createBuffer(Context ctx, size_t byte_size, AccessFlags access);
 void destroyBuffer(Context ctx, Buffer b);
-
 void* hostBufferSync(Context ctx, Buffer b, size_t byte_size, AccessFlags access);
-
 DeviceBuffer deviceBufferSync(Context ctx, Buffer b, size_t byte_size, AccessFlags access);
 
-Kernel loadKernelFromSource(Context ctx, const char* name, const char* source, size_t length);
-Kernel loadKernelFromFile(Context ctx, const char* name, const char* path);
 void destroyKernel(Context ctx, Kernel k);
 
-typedef struct {
-  size_t size;
-  void* value;
-} KernelArg;
-
-void launchKernel(
-  Context ctx, Kernel k,
-  const size_t* global_size, const size_t* local_size,
-  size_t arg_count, const KernelArg* args);
+#define KARG(val) { sizeof(val), &val }
+#define LARG(size) { size, NULL }
```

Comment:
I do not understand why these macros are moved here, are they meaningful for gap8?

Reply:
I reused ```KARG``` macro, probably moved ```LARG``` with it to mutual interface by inertia. Anyways, ```KARG``` is meaningful for GAP8,

Reply:
Same as for ```struct KernelArgs```, I have moved these two macros to the Ocl specific include